ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,922 @@
-        return all([isinstance(dim, Categorical) for dim in self.dimensions])
-
-    @property
-    def is_partly_categorical(self):
-        """Space contains any categorical dimensions"""
-        return any([isinstance(dim, Categorical) for dim in self.dimensions])
-
-    def distance(self, point_a, point_b):
-        """Compute distance between two points in this space.
-
-        Parameters
-        ----------
-        * `a` [array]
-            First point.
-
-        * `b` [array]
-            Second point.
-        """
-        distance = 0.
-        if len(self.dimensions) > 1:
-            for a, b, dim in zip(point_a, point_b, self.dimensions):
-                distance += dim.distance(a, b)
-                
-        if len(self.dimensions) == 1:
-             distance +=  self.dimensions[0].distance(point_a[0], point_b[0])
-
-        return distance
-
-    def lhs(self, n):
-        """ Returns n latin hypercube samples as a list of lists
-        """
-
-        samples = []
-        for i in range(self.n_dims):
-            lhs_perm = []
-            # Get evenly distributed samples from one dimension
-            lhs_aranged = self.dimensions[i].lhs_arange(n)
-            perm = np.random.RandomState(seed=42 + i).permutation(n)
-            for p in perm:  # Random permutate the order of the samples
-                lhs_perm.append(lhs_aranged[p])
-            samples.append(lhs_perm)
-        # Now we have a list of lists with samples for each dimension.
-        # We need to transpose this so that we get a list of lists with
-        # samples for all the dimensions
-        transposed_samples = []
-        for i in range(n):
-            row = []
-            for j in range(self.n_dims):
-                row.append(samples[j][i])
-            transposed_samples.append(row)
-        return transposed_samples
-    
-    # TODO: Add a R1 QRS sampling method here
+from abc import ABC, abstractmethod
+from typing import Iterable, List, Union
+
+import numbers
+import numpy as np
+import yaml
+
+
+from .transformers import CategoricalEncoder
+from .transformers import Normalize
+from .transformers import Identity
+from .transformers import Log10
+from .transformers import Pipeline
+
+# helper class to be able to print [1, ..., 4] instead of [1, '...', 4]
+
+
+class _Ellipsis:
+    def __repr__(self):
+
+def get_random_generator(
+    input: Union[int, float, np.random.RandomState, np.random.Generator, None]
+) -> np.random.Generator:
+    """Get a random generator from an input.
+
+    Parameters
+    ----------
+    * `input` [int, float, RandomState instance, Generator instance, or None (default)]:
+        Set random state to something other than None for reproducible
+        results.
+
+    Returns
+    -------
+    * `rng`: [Generator instance]
+       Random generator.
+    """
+    if input is None:
+        return np.random.default_rng()
+    elif isinstance(input, (int, float)):
```

Comment:
Looking at the docs for default_rng it looks like it does not accept floats as seeds (it states `seed
{None, int, array_like[ints], SeedSequence, BitGenerator, Generator}, optional`)

Reply:
I only allow for 10 000 different values, but since the use case is to ensure reproducibility, and RandomState is deprecated, I think that should be enough.